### PR TITLE
update all examples for 0.2.9

### PR DESCRIPTION
**Pull request by Lendemor**

### CODE REVIEW

File: qr-scanner/qr_scanner/component.py [hunk 1]
```diff
@@ -20,12 +20,15 @@ class QrScanner(NoSSRComponent):
...
-        return super().get_controlled_triggers() | {
-            "on_result": rx.EVENT_ARG,
-            "on_decode": rx.EVENT_ARG,
-            "on_error": BaseVar(name=rx.EVENT_ARG.name + "?.message", type_=str),
+
+        return {
+            **super().get_event_triggers(),
+            "on_result": lambda e0: [e0],
+            "on_decode": lambda e0: [e0],
+            "on_error": lambda e0: [BaseVar(name="_e0" + ".message")],
```
Comment (masenf) [inline qr-scanner/qr_scanner/component.py:30]:
```suggestion
            "on_error": lambda e0: [BaseVar(name="_e0?.message")],
```

File: quiz/quiz/quiz.py [hunk 2]
```diff
@@ -38,6 +40,10 @@ def submit(self):
         self.score = int(correct / total * 100)
         return rx.redirect("/result")
 
+    @rx.var
+    def percent_score(self):
+        return str(self.score) + "%"
```
Comment (masenf) [inline quiz/quiz/quiz.py:45]:
```suggestion
        return f"{self.score}%"
```

File: sales/sales/sales.py [hunk 3]
```diff
@@ -73,8 +112,12 @@ async def call_openai(self):
...
         )
         self.gen_response = False
-        self.email_content_data = response.choices[0].text  # save the data related to email_content
-        return rx.set_value("email_content", self.email_content_data) # update layout of email_content manually
+        self.email_content_data = response.choices[
+            0
+        ].text  # save the data related to email_content
+        return rx.set_value(
+            "email_content", self.email_content_data
+        )  # update layout of email_content manually
```
Comment (masenf) [inline sales/sales/sales.py:120]:
```suggestion
        # save the data related to email_content
        self.email_content_data = response.choices[0].text
        # update layout of email_content manually
        return rx.set_value("email_content", self.email_content_data)  
```

**Review by masenf — CHANGES_REQUESTED** [review]
couple of suggests

File: qr-scanner/qr_scanner/component.py (hunk 1 above)
Comment (Lendemor) [reply qr-scanner/qr_scanner/component.py:30]: What does the `?` do? 
It worked without it for me.

Comment (masenf) [reply qr-scanner/qr_scanner/component.py:30]: it doesn't deref the prop when the var is undefined or null. basically just avoids a runtime error if this gets called in some circumstances with a missing error.

i'm not sure if it's really necessary anymore, but when i was first prototyping this example app, i think there were some browsers that gave back a bogus error object if you denied camera permission or weren't running on localhost/with TLS.

Comment (masenf) [reply qr-scanner/qr_scanner/component.py:30]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

**Comment by Lendemor** [comment]
Applied all 3 of them :+1:

**Review by masenf — APPROVED** [review]

**Pull request merged**